File: tests/fixtures/cpp/src/audio.cpp
#include "audio.h"
#include "globals.h"
#include <cmath>

namespace audio {

float process_buffer(AudioBuffer* buf) {
    float sum_sq = 0.0f;
    for (uint32_t i = 0; i < buf->size; i++) {
        sum_sq += buf->samples[i] * buf->samples[i];
    }
    g_counter++;
    return sqrtf(sum_sq / buf->size);
}

AudioBuffer generate_sine(float freq) {
    AudioBuffer buf;
    buf.sample_rate = 44100;
    buf.size = 512;
    for (uint32_t i = 0; i < buf.size; i++) {
        buf.samples[i] = sinf(2.0f * 3.14159f * freq * i / buf.sample_rate);
    }
    return buf;
}

void apply_effect(AudioBuffer* buf, float gain) {
    for (uint32_t i = 0; i < buf->size; i++) {
        buf->samples[i] *= gain;
    }
}

} // namespace audio

File: tests/fixtures/rust/src/audio.rs
pub fn process_buffer(samples: &[f32]) -> f32 {
    let sum_sq: f32 = samples.iter().map(|s| s * s).sum();
    (sum_sq / samples.len() as f32).sqrt()
}

pub fn generate_sine(freq: f32) -> Vec<f32> {
    let sample_rate = 44100.0f32;
    (0..512)
        .map(|i| (2.0 * std::f32::consts::PI * freq * i as f32 / sample_rate).sin())
        .collect()
}

pub fn apply_effect(samples: &mut [f32], gain: f32) {
    for s in samples.iter_mut() {
        *s *= gain;
    }
}
